packages/jsii-reflect/lib/method.ts

```diff
@@ -0,0 +1,97 @@
+import jsii = require('jsii-spec');
+import { Assembly } from './assembly';
+import { Docs, Documentable } from './docs';
+import { Overridable } from './overridable';
+import { Parameter } from './parameter';
+import { Type } from './type';
+import { TypeReference } from './type-ref';
+import { TypeSystem } from './type-system';
+
+export class Method implements Documentable, Overridable {
+  constructor(
+    public readonly system: TypeSystem,
+    public readonly assembly: Assembly,
+    public readonly parentType: Type,
+    private readonly spec: jsii.Method) { }
+
+  public toString() {
+    return `method:${this.parentType.fqn}.${this.name}`;
+  }
+
+  /**
+   * The name of the method. Undefined if this method is a initializer.
+   */
+  public get name(): string {
+    if (this.initializer) {
+      return '<initializer>';
+    }
+
+    if (!this.spec.name) {
+      throw new Error(`Method does not have a name and is not an initializer`);
+    }
+
+    return this.spec.name;
+  }
+
+  /**
+   * The return type of the method (undefined if void or initializer)
+   */
+  public get returns(): TypeReference {
+    return new TypeReference(this.system, this.spec.returns);
```

Comment:
Feels like this could (should?) be cached, maybe.

Reply:
Since everything is immutable, I'll leave caching for a later stage. There are many opportunities for caching all across this library since it's all lazy.